terminal.go

```diff
@@ -16,10 +17,14 @@ import (
 type Terminal struct {
 	// Use this for any output to the screen/console so the required \r are added in raw mode
 	// the prompt and command edit is refresh as needed when input comes in.
-	Out         io.Writer
+	Out io.Writer
+	// Cancellable context after Open(). Use it to cancel the terminal reading or check for done.
+	Context     context.Context //nolint:containedctx // To avoid Open() returning 4 values.
+	Cancel      context.CancelFunc
```

Comment:
Maybe ContextCancel, Cancel is a bit unclear

Reply:
nah I think `Cancel()` is pretty "go"